clients/python/moondream/cloud_vl.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+import urllib.request
+import base64
+from PIL import Image
+from io import BytesIO
+from typing import Union, Optional, Literal
+
+
+from .types import (
+    VLM,
+    CaptionOutput,
+    EncodedImage,
+    QueryOutput,
+    DetectOutput,
+    SamplingSettings,
+)
+
+
+class CloudVL(VLM):
+    @classmethod
+    def from_api_key(
+        cls, api_key: str, api_url: str = None, version: str = "v1"
+    ) -> "CloudVL":
+        """Initialize a CloudVL instance with an API key.
+
+        Args:
+            api_key (str): The API key for authentication
+            api_url (str, optional): Custom API URL. Defaults to None.
+            version (str, optional): API version. Defaults to "v1".
+
+        Returns:
+            CloudVL: An initialized CloudVL instance
+        """
+        return cls(api_key=api_key, api_url=api_url, version=version)
+
+    def __init__(self, api_key: str = None, api_url: str = None, version: str = "v1"):
+        self.api_key = api_key or os.getenv("MOONDREAM_API_KEY")
+        if not self.api_key:
+            raise ValueError(
+                "API key is required. Set MOONDREAM_API_KEY env var or pass api_key"
+            )
+
+        self.api_url = api_url or f"https://api.moondream.ai/{version}"
```

Comment:
Hard-code to the URL for now. We can add support for customizing the endpoint when we need it.